.github/workflows/build-docs.yml

```
@@ -0,0 +1,15 @@
+name: Build docs
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+jobs:
+  build-docs:
+    runs-on: ubuntu-18.04
+    steps:
+      - uses: actions/checkout@v2
+      - uses: ammaraskar/sphinx-action@master
+        with:
+          pre-build-command: "pip install Sphinx==4.2.0 recommonmark==0.7.1"
+          docs-folder: "docs/"
```

The current setup would not treat warnings as errors, most of the problems with our docs building in the past were in most cases informed as warnings, but they were ignored, eg. missing entry in the index.  We should add "-W" option to enable fatal warnings, `--keep-going` to not stop after the first warning and optionally `-n` which would try to check for missing references
```suggestion
          build-command: 'make html SPHINXOPTS="-W --keep-going -n"'
          docs-folder: "docs/"
```

---

Hi, 
Before raising pull request, I thoroughly checked for all the options to make warning as error. 
I have checked on adding `-W` but it does not work.
To solve it I had already raised issue.
- [Add -W support](https://github.com/ammaraskar/sphinx-action/issues/34)
- [Add number of warning as step output](https://github.com/ammaraskar/sphinx-action/issues/35)

Even with one issue resolved, we can enable warning as error.

I also looked into catching `warning` as they bubble as Github warning, but i dont see any reference of any such `workflow commands`
If you find any resources/approach, please share. I will implement it.
Any thoughts?

---

@ankusharya I think that maybe problems with log files might be also a problem with running with act. We should check if it behaves the same in GH Actions environments in the same way. I don't have a large experience with this tool, but even then I came to corner cases where build was failing locally with act and worked well in the CI